File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.js
function solution(n) {
    let cnt = 1
    let ssum = 0
    let start = 1
    for (i=1; i < n+1; i++) {
        if (ssum === n) {
            cnt++
        }
        ssum += i
        while (ssum > n) {
            ssum -= start++
        }

    }
    return cnt;
}

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    total_count = 0
    current_sum = 0
    start = 1
    for end in range(1, n + 1):
        current_sum += end
        while current_sum > n:
            current_sum -= start
            start += 1
        
        if current_sum == n:
            total_count += 1
    
    return total_count